pkg/network/setup/configstate.go

```diff
@@ -45,40 +45,57 @@ func NewConfigState(cc cacheCreator, vmiUID string) ConfigState {
 //
 // The discovery step can be executed repeatedly with no limitation.
 // The configuration step is allowed to run only once. Any attempt to run it again will cause a critical error.
-func (c ConfigState) Run(podIfaceName string, discoverFunc func() error, configFunc func() error) error {
-	state, err := c.read(podIfaceName)
-	if err != nil {
-		return err
-	}
+func (c ConfigState) Run(nics []podNIC, discoverFunc func(*podNIC) error, configFunc func(*podNIC) error) error {
+	var pendingNICs []podNIC
+	for _, nic := range nics {
```

Comment:
is there anything preventing us from using one for loop?

IIUC, if after the first loop you continue with `pendingNICs` and not `nics` then it should be more straight forward (unless you plan to change this in next commit?).